node/src/node_configurator/node_configurator_standard.rs

```diff
@@ -988,4 +993,81 @@ mod tests {
         let expected = ExternalData::new(DEFAULT_CHAIN, NeighborhoodModeLight::ZeroHop, None);
         assert_eq!(result, expected)
     }
+
+    fn check_data_directory_combinations(
+        chain_opt: Option<&str>,
+        data_directory_opt: Option<&str>,
+        expected: Option<&str>,
+    ) {
+        let home_dir = PathBuf::from("/home/cooga");
+        let standard_data_dir = PathBuf::from("/home/cooga/.local");
+
+        let args = match (chain_opt, data_directory_opt) {
+            (Some(chain_opt), Some(data_directory_opt)) => ArgsBuilder::new()
+                .param("--chain", chain_opt)
+                .param("--real-user", "999:999:/home/cooga")
+                .param("--data-directory", data_directory_opt),
+            (None, Some(data_directory_opt)) => ArgsBuilder::new()
+                .param("--chain", "polygon-mainnet")
```

Comment:
Since the chain isn't specified in the function parameters, shouldn't it be left out here? If you leave it out here and the test fails, I think that would bear looking into.

Reply:
chain is going into `args_vec` and passed into `server_initializer_collected_params` function